metricflow/model/semantics/metric_lookup.py

```diff
@@ -105,9 +105,10 @@ def measures_for_metric(
             )
             spec = MetricInputMeasureSpec(
                 measure_spec=measure_spec,
-                constraint=WhereFilterSpec.create_from_where_filter(
-                    where_filter=input_measure.filter,
-                    column_association_resolver=column_association_resolver,
+                constraint=input_measure.filter.transform(
+                    ConvertToWhereSpec(
+                        column_association_resolver=column_association_resolver,
+                    )
```

Comment:
This seems silly given that below we just do `ConvertToWhereSpec(...).transform(filter)` and it also works.

Reply:
Up for debate, but I actually like the `starting_object.transform(transform_function)` syntax. It flows more naturally when reading.

Reply:
I'm suggesting this:

`new_object_type = object_converter.transform(original_object)`

This is also fine, although putting the operation on the instance suggests that in-place mutation might be happening:

`new_object_type = original_object.transform(transform_function)`

This is what we have in this PR:

```
new_object_type = object_converter.transform(original_object)

...

another_new_object = another_original_object.transform(object_converter)
```

We probably shouldn't use both constructions in the same file right out of the gate.

But none of this matters much.

Reply:
> We probably shouldn't use both constructions in the same file right out of the gate.

Agree - think I found those and made it consistent.